INSTALL.md

```diff
@@ -0,0 +1,142 @@
+         ));
+         ^
+    ./include/cudarray/nnet/cudnn.hpp:85:44: note: in definition of macro ‘CUDNN_CHECK’
+     #define CUDNN_CHECK(status) { cudnn_check((status), __FILE__, __LINE__); }
+ 
+    compilation terminated due to -Wfatal-errors.
+    make: *** [src/nnet/cudnn.o] Error 1
+    ```
+
+# DeepPy
+
+1. Download DeepPy:
+
+    ```
+    $ git clone https://github.com/andersbll/deeppy.git
+    ```
+
+1. Copy the `deeppy` module direcotry into the Neural Artistic Style directory:
+
+    ```
+    $ cp -a deeppy/deeppy neural_artistic_style
+    ```
+
+# Pretrained VGG 19 Model
+
+1. Download the pretrained VGG 19 model:
+
+    ```
+    $ cd neural_artistic_style
+    $ wget http://www.vlfeat.org/matconvnet/models/imagenet-vgg-verydeep-19.mat
+    ```
+
+The model size is around 510 MB.
+
+# Troubleshooting
+
+## Out of Memory Issue
+
+If you get the out of memory error messages when executing
+neural_artistic_style.py (Issue [#26](https://github.com/andersbll/neural_artistic_style/issues/26)):
+
+    ```
+    Traceback (most recent call last):
+      File "neural_artistic_style.py", line 138, in <module>
+        run()
+      File "neural_artistic_style.py", line 130, in run
+        cost = np.mean(net.update())
+      File "neural_artistic_style/style_network.py", line 130, in update
+        next_x = layer.fprop(next_x)
+      File "neural_artistic_style/deeppy/feedforward/convnet_layers.py", line 71, in fprop
+        poolout = self.pool_op.fprop(x)
+      File "/usr/local/lib/python2.7/dist-packages/cudarray-0.1.dev-py2.7-linux-x86_64.egg/cudarray/nnet/pool.py", line 34, in fprop
+        poolout = ca.empty(poolout_shape, dtype=imgs.dtype)
+      File "/usr/local/lib/python2.7/dist-packages/cudarray-0.1.dev-py2.7-linux-x86_64.egg/cudarray/cudarray.py", line 246, in empty
+        return ndarray(shape, dtype=dtype)
+      File "/usr/local/lib/python2.7/dist-packages/cudarray-0.1.dev-py2.7-linux-x86_64.egg/cudarray/cudarray.py", line 36, in __init__
+        self._data = ArrayData(self.size, dtype, np_data)
+      File "cudarray/wrap/array_data.pyx", line 16, in cudarray.wrap.array_data.ArrayData.__init__ (./cudarray/wrap/array_data.cpp:1401)
+      File "cudarray/wrap/cudart.pyx", line 12, in cudarray.wrap.cudart.cudaCheck (./cudarray/wrap/cudart.cpp:763)
+    ValueError: out of memory
+    ```
+                                                                                                                                         
+Here are some solutions:
+
+ 1. Use GPU with larger memory.
+ 1. Use smaller input and style images.
+ 1. Use CPU instead of GPU.  Set `CUDARRAY_BACKEND` to 'numpy' as workaround.
```

Comment:
As seen in the different issues, a warning should be displayed about using the CPU. The most important delta I read about was 10 days on CPU, 30 min on GPU.